prusti-tests/tests/verify/pass/equality/enum-empty.rs

```diff
@@ -0,0 +1,14 @@
+use prusti_contracts::*;
+
+#[derive(Clone,PartialEq,Eq)]
+enum A {
+}
```

Comment:
This is an interesting case, equivalent to [`Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html). You can never construct a value of this type, so a method taking `A`/`Infallible` as arguments can never be called. Maybe we should report this as a precondition failure.

Also relevant about `match` on this: https://stackoverflow.com/questions/47589690/what-does-a-match-on-an-empty-enum-return#47589799

Reply:
Another example is `Void`, which we also already have in our test suite. Having an instance of `Void` is equivalent to being in a contradictory state. Therefore, for your `test` function, we should be able to verify `#[ensures(false)]`. However, I would be fine if this is not implemented yet as part of this PR because we are already discussing how to support this with @tillarnold.

Reply:
I agree that we leave that corner case to @vakaras and @tillarnold.
We will keep the test to ensure that Prusti does not falsely assume that enums have at least one variant.